addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScanRule.java

```diff
@@ -105,28 +109,41 @@ public static Optional<String> getHost(Element element, String origin) {
                     CommonAlertTag.OWASP_2021_A05_SEC_MISCONFIG,
                     CommonAlertTag.OWASP_2017_A06_SEC_MISCONFIG);
 
+    private final TrustedDomains trustedDomains = new TrustedDomains();
+
     @Override
     public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        trustedDomains.update(getConfig().getString(TrustedDomains.TRUSTED_DOMAINS_PROPERTY, ""));
+        // trustedDomains.add(new SameOriginPolicyTrust(msg.getRequestHeader().getURI()));
+        try {
+            trustedDomains.add(new RegexTrust(msg.getRequestHeader().getURI().getHost()));
+        } catch (URIException e) {
+            // Ignore
+        }
```

Comment:
Given the current Unit Tests the rule actually allows something more relaxed than SOP.